src/frontend/src/planner/select.rs

```diff
@@ -101,56 +101,39 @@ impl Planner {
             Condition::with_expr(where_clause)
                 .group_by::<_, 3>(|expr| match expr {
                     ExprImpl::Subquery(_) => 0,
-                    ExprImpl::FunctionCall(func_call)
-                        if func_call.get_expr_type() == ExprType::Not
-                            && matches!(func_call.inputs()[0], ExprImpl::Subquery(_)) =>
-                    {
-                        1
-                    }
+                    ExprImpl::FunctionCall(func_call) => match func_call.get_expr_type() {
+                        ExprType::In if matches!(func_call.inputs()[1], ExprImpl::Subquery(_)) => 0,
+                        ExprType::Not => {
+                            let expr = &func_call.inputs()[0];
+                            match expr {
+                                ExprImpl::Subquery(_) => 1,
+                                ExprImpl::FunctionCall(func)
+                                    if func.get_expr_type() == ExprType::In
+                                        && matches!(func.inputs()[1], ExprImpl::Subquery(_)) =>
+                                {
+                                    1
+                                }
+                                _ => 2,
+                            }
+                        }
+                        _ => 2,
```

Comment:
In the description of this PR, it is better to be clear about the limitation:
in-subquery that goes into `others` cannot be handled by `substitute_subqueries` yet.